Sources/Hummingbird/Middleware/LogRequestMiddleware.swift

```diff
@@ -12,32 +12,103 @@
 //
 //===----------------------------------------------------------------------===//
 
+import HTTPTypes
 import Logging
 
 /// Middleware outputting to log for every call to server
 public struct LogRequestsMiddleware<Context: BaseRequestContext>: RouterMiddleware {
+    /// Header filter
+    public enum HeaderFilter: Sendable, ExpressibleByArrayLiteral {
+        public typealias ArrayLiteralElement = HTTPField.Name
+
+        /// ExpressibleByArrayLiteral requirement
+        public init(arrayLiteral elements: ArrayLiteralElement...) {
+            self = .some(elements)
+        }
+
+        case none
+        case all(except: [HTTPField.Name] = [])
+        case some([HTTPField.Name])
+    }
+
     let logLevel: Logger.Level
-    let includeHeaders: Bool
+    let includeHeaders: HeaderFilter
+    let redactHeaders: [HTTPField.Name]
 
-    public init(_ logLevel: Logger.Level, includeHeaders: Bool = false) {
+    public init(_ logLevel: Logger.Level, includeHeaders: HeaderFilter = .none, redactHeaders: [HTTPField.Name] = []) {
         self.logLevel = logLevel
         self.includeHeaders = includeHeaders
+        // only include headers in the redaction list if we are outputting them
+        self.redactHeaders = switch includeHeaders {
+        case .all(let except):
+            // don't include headers in the except list
+            redactHeaders.filter { header in except.first { $0 == header } == nil }
+        case .some(let included):
+            // only include headers in the included list
+            redactHeaders.filter { header in included.first { $0 == header } != nil }
```

Comment:
```suggestion
        case .all(let exceptions):
            // don't include headers in the except list
            redactHeaders.filter { header in !exceptions.contains(header) }
        case .some(let included):
            // only include headers in the included list
            redactHeaders.filter { header in included.contains(header) }
```

Reply:
I prefer `except`. It reads better. All headers except. But I'll change to contains